src/plugins/factoids/factoidsPlugin.js

```diff
@@ -6,16 +6,81 @@ function parseMsg(msg) {
     return { key: null, nick: null };
   }
   const [key, nick] = msg.command.command.split('@').map((x) => x.trim());
-  return { key, nick: nick || null };
+
+  const findParts = key.split(/^find[ ]*/);
+  if (findParts.length > 1) {
+    return { key: null, nick: nick || null, find: findParts[1] };
+  }
+
+  return { key, nick: nick || null, find: null };
 }
 
+const findCommand = (query) => {
+  const DISPLAY_LIMIT = 5;
+  const MATCH_LIMIT = DISPLAY_LIMIT + 1;
+  const needle = query.toLowerCase();
+  const points = Object.create(null);
+
+  for (const [key, fact] of Object.entries(facts)) {
+    const addPoints = (amount) => {
+      points[key] = points[key] || 0;
+      points[key] += amount;
+    };
+    const keyIsPrefix = key.toLowerCase().startsWith(needle);
+    const matchesKey = key.toLowerCase().includes(needle);
+    const matchesValue = fact.value && fact.value.toLowerCase().includes(needle);
+
+    if (matchesKey) {
+      addPoints(10);
+    }
+    if (keyIsPrefix) {
+      addPoints(5);
+    }
+    if (matchesValue) {
+      // A slight boost for popular factoids.
+      addPoints(Math.log(fact.value.popularity || 1));
+    }
+  }
+
+  const matches = Object.keys(points)
+    .sort((a, b) => points[b] - points[a])
+    .slice(0, MATCH_LIMIT);
+
+  if (matches.length === 0) {
+    return `I couldn't find any factoid triggers/messages with that text.`;
+  }
+  if (matches.length === 1) {
+    const [first] = matches;
+    return `I found a matching factoid: !${first}`;
+  }
+
+  const andMore = matches.length === MATCH_LIMIT;
+  const items = matches.slice(0, DISPLAY_LIMIT).map((k) => `!${k}`);
+  const last = items.pop();
+  const tail = andMore ? `and more` : `and ${last}`;
+  const formatted = [...items, tail].join(', ');
+
+  return `I found matching factoids: ${formatted}`;
```

Comment:
```suggestion
  return `Found: ${formatted}`;
```
so the list can fit more things in it

Reply:
Went with lowercase 'found' and no punctuation due to the mention.